src/main/java/az/edu/strangers/service/FamilyServiceImpl.java

```diff
@@ -101,31 +108,47 @@ public FamilyDto bornChild(FamilyDto familyDto, String masculineName, String fem
 
     @Override
     public Family adoptChild(Family family, Human child) {
-        return null;
+        if(family==null||child==null) return null;
+        family.addChild(child);
+        return family;
     }
 
     @Override
     public void deleteAllChildrenOlderThen(Integer age) {
+        List<Family> familyList=getAllFamilies();
+        familyList.forEach(family -> {
+            List<Human> olderChildren=family.getChildren().stream().filter(child->child.getYear()<=age)
+                    .collect(Collectors.toList());
+            family.setChildren(olderChildren);
+          //  familyDao.updateFamily(familyDao);
+        });
 
     }
 
     @Override
     public Long count() {
-        return null;
+        return (long)familyDao.getAllFamilies().size();
     }
 
     @Override
     public Optional<Family> getFamilyById(Long index) {
-        return Optional.empty();
+        List<Family> familyList=familyDao.getAllFamilies();
+        return index==null||familyList.size()<=index? Optional.empty():
+        Optional.of(familyList.get(index.intValue()));
     }
 
     @Override
-    public List<Pet> getPets(Long index) {
-        return null;
+    public Optional<Pet> getPets(Long index) {
+        List<Family> familyList=familyDao.getAllFamilies();
+        return index>= familyList.size()? Optional.empty():
+                Optional.of(familyList.get(index.intValue()).getPet());
     }
 
     @Override
-    public void addPet(Long index, Pet pet) {
-
+    public boolean addPet(final Long index,final Pet pet) {
+        List<Family> familyList=familyDao.getAllFamilies();
+        if (familyList.size()<=index||index<0) return false;
+        familyList.get(index.intValue());
+        return true;
```

Comment:
this method is not properly written, the goal here was to add a new animal to the family